src/SDK/MonitoringApplianceEmulator/Models/SmartDetectorRunner.cs

```diff
@@ -145,47 +159,61 @@ public async Task RunAsync(HierarchicalResource targetResource, List<ResourceIde
                     int currentRunNumber = 1;
                     for (var currentTime = startTimeRange; currentTime <= endTimeRange; currentTime = currentTime.Add(analysisCadence))
                     {
-                        this.Tracer.TraceInformation($"Start analysis, end of time range: {currentTime}");
-
-                        ExtendedDateTime.SetEmulatedUtcNow(currentTime);
-                        var analysisRequest = new AnalysisRequest(targetResourcesForDetector, analysisCadence, null, this.analysisServicesFactory, stateRepository);
-
-                        // Run the detector in a different context by using "Task.Run()". This will prevent the detector execution from blocking the UI
-                        List<SmartDetectors.Alert> newAlerts = await Task.Run(() => this.smartDetector.AnalyzeResourcesAsync(
-                            analysisRequest,
-                            this.Tracer,
-                            cancellationToken));
-
-                        var smartDetectorExecutionRequest = new SmartDetectorExecutionRequest
-                        {
-                            ResourceIds = targetResourcesIds,
-                            SmartDetectorId = this.smartDetectorManifest.Id,
-                            Cadence = analysisCadence,
-                        };
-
-                        var lazyResourceToWorkspaceResourceIdMapping = new Lazy<Task<Dictionary<ResourceIdentifier, ResourceIdentifier>>>(() => this.GetResourceToWorkspaceResourceIdMappingAsync(targetResourcesForDetector, cancellationToken));
-
-                        foreach (var newAlert in newAlerts)
+                        this.PageableLog = await this.logArchive.GetLogAsync(this.GetValidLogName(currentTime), 50);
```

Comment:
>= [](start = 41, length = 1)

I think we should encapsulate the GetLogAsync operation in a Task.Run, like we aredoing for this.smartDetector.AnalyzeResourcesAsync, so it won't stuck the UI thread

Reply:
this is not running on the UI thread

---
In reply to: [231552409](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/13#discussion_r231552409) [](ancestors = 231552409)